### PR TITLE
Rosalina Teleport downb

### PR DESCRIPTION
**Pull request by chrispo-git**
Rosalina downb now acts as a teleport, swapping her and Luma's position.
Frames 17-24 - invulnerable (teleporting)
Frame 38 - Actionable

Rosalina cannot use downb for 300 frames (5 seconds) afterwards, and her being able to use it again is indicated by a flash on her hands and on the HUD. If you're too far away, Rosalina cannot teleport.

https://user-images.githubusercontent.com/69305550/177057739-f0bee2e4-5758-48f0-8e09-984b94f8e265.mp4

### CODE REVIEW

File: fighters/rosetta/src/opff.rs
Comment (WuBoytH) [inline fighters/rosetta/src/opff.rs]: Weapons can have their own specific fighter frames, you can change the attribute to this `#[weapon_frame( agent = WEAPON_KIND_ROSETTA_TICO )]`

**Review by WuBoytH — CHANGES_REQUESTED** [review]
Look at my comment about Luma's OPFF. Other than that, it should need playtesting (and a second opinion from someone else) for further adjustments before approval.

**Comment by jobrien97** [comment]
Please change the indicator when teleport is recharged to our common gimmick flash effect found in common/src/opff/gimmick.rs

File: fighters/rosetta/src/opff.rs
Comment (WuBoytH) [inline fighters/rosetta/src/opff.rs]: you can just do `VarModule::dec_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN);` to decrement this var by 1.

File: fighters/rosetta/src/opff.rs
Comment (WuBoytH) [inline fighters/rosetta/src/opff.rs]: I would appreciate that you don't nest VarModule calls for readability purposes.
`let rosa_x = ...;`, `let tico_x = ...;`, etc. Please check this for the rest of the function as well.

**Review by WuBoytH — CHANGES_REQUESTED** [review]
Just a few smaller, organizational changes are needed before we can merge this in. Once it *does* get merged in, I may try and reimplement this as a status script instead.

**Review by WuBoytH — APPROVED** [review]
LGTM

**Pull request merged**
